src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -162,7 +162,11 @@ private void CreateNodeCaches()
             {
                 if (_compilationModuleGroup.ContainsType(type))
                 {
-                    if (type.IsCanonicalSubtype(CanonicalFormKind.Any))
+                    if (type.IsCanonicalDefinitionType(CanonicalFormKind.Any))
+                    {
+                        return new CanonicalDefinitionEETypeNode(this, type);
```

Comment:
ProjectX HostedCompilation.cs in RootRequiredNodes(). Again, codebase that i'm not really familiar with. @davidwrighton mentioned there's a good reason why it asks for constructed types.